cms/djangoapps/contentstore/views/tests/test_organizations.py

```diff
@@ -32,3 +38,90 @@ def test_organization_list(self):
         self.assertEqual(response.status_code, 200)
         org_names = json.loads(response.content.decode('utf-8'))
         self.assertEqual(org_names, self.org_short_names)
+
+
+class TestOrganizationsForLibraries(TestCase):
+    """
+    Verify who is allowed to create Libraries.
+
+    This uses some low-level implementation details to set up course creator and
+    org staff data, which should be replaced by API calls.
+
+    The behavior of this call depends on two FEATURES toggles:
+
+    * ENABLE_ORGANIZATION_STAFF_ACCESS_FOR_CONTENT_LIBRARIES
+    * ENABLE_CREATOR_GROUP
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.library_author = UserFactory(is_staff=False)
+        cls.org_short_names = ["OrgStaffOrg", "CreatorOrg", "RandomOrg"]
+        cls.orgs = {}
+        for index, short_name in enumerate(cls.org_short_names):
+            cls.orgs[short_name] = add_organization(organization_data={
+                'name': 'Test Organization %s' % index,
+                'short_name': short_name,
+                'description': 'Testing Organization %s Description' % index,
+            })
+
+        # Our user is an org staff for OrgStaffOrg
+        OrgStaffRole("OrgStaffOrg").add_users(cls.library_author)
+
+        # Our user is also a CourseCreator in CreatorOrg
+        creator = CourseCreator.objects.create(
+            user=cls.library_author,
+            state=CourseCreator.GRANTED,
+            all_organizations=False,
+        )
+        # The following is because signal catching code assumes that changes are
+        # only ever done through Django admin, and I don't have time to untangle
+        # that right now.
+        creator.admin = UserFactory(is_staff=True)
```

Comment:
Interesting. I don't understand how this line would affect signal handlers. Could you expand the comment a bit, just enough so that if this test broke somewhere down the line, we'd have enough of a hint to figure out how you got it working here?